x-pack/test/apm_api_integration/tests/transactions/transactions_groups_alerts.spec.ts

```diff
@@ -0,0 +1,282 @@
+      },
+      {
+        name: 'GET /api/failed/request',
+        duration: 100,
+        type: 'request',
+      },
+    ];
+    before(async () => {
+      const serviceGoProdInstance = apm
+        .service({ name: serviceName, environment: 'production', agentName: 'go' })
+        .instance('instance-a');
+
+      await synthtraceEsClient.index([
+        timerange(start, end)
+          .interval('1m')
+          .rate(5)
+          .generator((timestamp) => {
+            return transactions.map(({ name, duration, type }) => {
+              return serviceGoProdInstance
+                .transaction({ transactionName: name, transactionType: type })
+                .timestamp(timestamp)
+                .duration(duration)
+                .success();
+            });
+          }),
+        timerange(start, end)
+          .interval('1m')
+          .rate(1)
+          .generator((timestamp) => {
+            return transactions.map(({ name, duration, type }) => {
+              return serviceGoProdInstance
+                .transaction({ transactionName: name, transactionType: type })
+                .timestamp(timestamp)
+                .duration(duration)
+                .failure();
+            });
+          }),
+      ]);
+    });
+
+    describe('Transaction groups with avg transaction duration alerts', () => {
+      let ruleId: string;
+
+      before(async () => {
+        const createdRule = await createApmRule({
+          supertest,
+          name: `Latency threshold | ${serviceName}`,
+          params: {
+            serviceName,
+            transactionType: 'request',
+            transactionName: 'GET /api/request/avg',
+            windowSize: 99,
+            windowUnit: 'h',
+            threshold: 99,
+            aggregationType: AggregationType.Avg,
+            environment: 'production',
+          },
+          ruleTypeId: ApmRuleType.TransactionDuration,
+        });
+
+        ruleId = createdRule.id;
+        await waitForActiveAlert({ ruleId, esClient, log });
+      });
+
+      after(async () => {
+        await supertest.delete(`/api/alerting/rule/${ruleId}`).set('kbn-xsrf', 'true');
+      });
+
+      it('returns the correct number of alert counts', async () => {
+        const txGroupsTypeRequest = await getTransactionGroups({
+          query: { transactionType: 'request' },
+        });
+
+        expect(txGroupsTypeRequest.hasActiveAlerts).to.be.equal(true);
+
+        ['GET /api/failed/request', 'GET /api/request/p95', 'GET /api/request/p99'].map(
+          (txName) => {
+            expect(
+              txGroupsTypeRequest.transactionGroups.find(({ name }) => name === txName)
+            ).to.not.have.property('alertsCount');
+          }
+        );
+
+        ['GET /api/request/avg'].map((txName) => {
+          const tx = txGroupsTypeRequest.transactionGroups.find(({ name }) => name === txName);
+          expect(tx).to.have.property('alertsCount', 1);
+        });
```

Comment:
What about this? I also think it would be a little nicer to always have `alertsCount` defined.

```suggestion
const expected = txGroupsTypeRequest.transactionGroups.map(
  ({ name, alertsCount }) => ({name, alertsCount})
);

expect(expected).to.eql([
  { name: 'GET /api/failed/request', alertsCount: 0 },
  { name: 'GET /api/request/p95', alertsCount: 0 },
  { name: 'GET /api/request/p99', alertsCount: 0 },
  { name: 'GET /api/request/avg', alertsCount: 1 },
])

```

Reply:
thanks for the suggestion, will update it